index.js

```diff
@@ -160,10 +165,20 @@ Filter.prototype.build = function() {
             return fs.unlinkSync(outputPath);
           } case 'change': {
             instrumentation.change++;
-            return plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            var changeOperation = plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            if (plugin.async) {
```

Comment:
I'm not totally sure about that, but lets see how it plays out. I suspect provide concurrency option, without a good throttling will lead to issues. But we can always start with the throttling in broccoli-persistent-filter and see how it works before moving it into this library.